### PR TITLE
[NETBEANS-451] search for git revision instead of mercurial one.

### PR DESCRIPTION
**Pull request by nigjo**
Changed to search for ".git/HEAD" instead of ".hg/dirstate"

### CODE REVIEW

**Comment by lolo101** [comment]
Why not changing the class name instead of adding a comment explaining why the name is no more relevant ?

Prefer using uppercase 'L' to denote long type (`0L` is more readable than `0l`)

File: nbbuild/antsrc/org/netbeans/nbbuild/HgId.java [hunk 1]
```diff
@@ -53,33 +56,43 @@ public void setProperty(String property) {
...
+                String revLink = line.substring(line.indexOf(':')+1).trim();
+
+                Path revPath = headroot.getParent().resolve(revLink);
+                if(Files.isRegularFile(revPath) && Files.size(revPath) > 0l) {
+                    List<String> revlines = Files.readAllLines(revPath);
+                    String revline = revlines.get(0);
+                    if(revline.length()>=12){
+                        id = revline.substring(0, 12);
+                    } else {
+                        log("no content in " + revPath, Project.MSG_WARN);
```
Comment (lolo101) [inline nbbuild/antsrc/org/netbeans/nbbuild/HgId.java]: That log might be misleading. Including `revline` to the log adds precious context.
For instance :
```java
log("content of " + revPath + " could not be parsed as an ID : \"" + revline + "\"");
```

File: nbbuild/antsrc/org/netbeans/nbbuild/HgId.java [hunk 2]
```diff
@@ -53,33 +56,43 @@ public void setProperty(String property) {
...
         if (file == null || property == null) {
             throw new BuildException("define file and property");
         }
-        File dirstate = null;
-        for (File root = file; root != null; root = root.getParentFile()) {
-            File ds = new File(new File(root, ".hg"), "dirstate");
-            if (ds.isFile()) {
-                dirstate = ds;
+        Path headroot = null;
+        for (Path root = file.toPath(); root != null; root = root.getParent()) {
```
Comment (lolo101) [inline nbbuild/antsrc/org/netbeans/nbbuild/HgId.java]: That loop should be placed in a separate method :
```java
Path headroot = findHead(file.toPath());
...
private Path findHead(Path root) {
    for (; root != null; root = root.getParent()) {
        Path headpath = root.resolve(".git/HEAD");
        if (Files.isRegularFile(headpath)) {
            return headpath;
        }
    }
    return null;
}
```

**Comment by nigjo** [comment]
> Why not changing the class name instead of adding a comment explaining why the name is no more relevant ?
Changing the name of this ant-task will lead to more changes in the build scripts itself. This pull-request just changes the source of the generated ID but not the "behavior" of the ant-task.

File: nbbuild/antsrc/org/netbeans/nbbuild/HgId.java (hunk 1 above)
Comment (nigjo) [reply nbbuild/antsrc/org/netbeans/nbbuild/HgId.java]: The git-commit ID is longer than 12 characters (see git docs). so the "revline" is always longer, but has to be at least 12 characters.

Comment (nigjo) [reply nbbuild/antsrc/org/netbeans/nbbuild/HgId.java]: I don't see a benefit of another log line. the revline should always be a commit-id (again, see git docs) or something not git compliant.

Comment (lolo101) [reply nbbuild/antsrc/org/netbeans/nbbuild/HgId.java]: Agree with the "at least 12 chars". Actually I removed my comment about that before noticing your answer to it.
I dont suggest adding a new log, but adding context to the existing one.
The benefit is for the user who may have a corrupt (but non empty) "revPath" and would be puzzled by a "no content" log.

**Comment by JaroslavTulach** [comment]
Nice work, I also wanted to implement the support for `.git`, so thanks for taking care of that.

**Comment by matthiasblaesing** [comment]
I see general agreement and came to the same conclusion as the other reviewers - this looks good and a testbuild resultat in the expected result (revision is part of product version). I'm merging this to master.

Thank you!

**Pull request merged**